packages/compass-indexes/src/components/regular-indexes-table/regular-indexes-table.tsx

```diff
@@ -50,96 +37,73 @@ export const RegularIndexesTable: React.FunctionComponent<
+      ],
+      actions: index.name !== '_id_' && index.extra.status !== 'inprogress' && (
+        <IndexActions
+          index={index}
+          serverVersion={serverVersion}
+          onDeleteIndex={onDeleteIndex}
+          onHideIndex={onHideIndex}
+          onUnhideIndex={onUnhideIndex}
+        ></IndexActions>
+      ),
+      details: <IndexKeysBadge keys={index.fields} />,
+    };
+  });
 
   return (
-    <IndexesTable<IndexDefinition>
+    <IndexesTable
       data-testid="indexes"
       aria-label="Indexes"
+      canModifyIndex={canModifyIndex}
       columns={columns}
-      data={indexes}
-    >
-      {({ datum: index }) => {
-        return (
-          <Row
-            key={index.name}
-            data-testid={`index-row-${index.name}`}
-            className={rowStyles}
-          >
-            <Cell data-testid="index-name-field" className={cellStyles}>
-              {index.name}
-            </Cell>
-            <Cell data-testid="index-type-field" className={cellStyles}>
-              <TypeField type={index.type} extra={index.extra} />
-            </Cell>
-            <Cell data-testid="index-size-field" className={cellStyles}>
-              <SizeField size={index.size} relativeSize={index.relativeSize} />
-            </Cell>
-            <Cell data-testid="index-usage-field" className={cellStyles}>
-              <UsageField usage={index.usageCount} since={index.usageSince} />
-            </Cell>
-            <Cell data-testid="index-property-field" className={cellStyles}>
-              <PropertyField
-                cardinality={index.cardinality}
-                extra={index.extra}
-                properties={index.properties}
-              />
-            </Cell>
-            {/* Index actions column is conditional */}
-            {canModifyIndex && (
-              <Cell data-testid="index-actions-field" className={cellStyles}>
-                {index.name !== '_id_' && index.extra.status !== 'inprogress' && (
-                  <div
-                    className={cx(indexActionsCellStyles, 'index-actions-cell')}
-                  >
-                    <IndexActions
-                      index={index}
-                      serverVersion={serverVersion}
-                      onDeleteIndex={onDeleteIndex}
-                      onHideIndex={onHideIndex}
-                      onUnhideIndex={onUnhideIndex}
-                    ></IndexActions>
-                  </div>
-                )}
-              </Cell>
-            )}
-            <Row>
-              <Cell
-                className={cx(nestedRowCellStyles, cellStyles)}
-                colSpan={canModifyIndex ? 6 : 5}
-              >
-                <IndexKeysBadge keys={index.fields} />
-              </Cell>
-            </Row>
-          </Row>
-        );
-      }}
-    </IndexesTable>
+      data={data}
+      onSortTable={(column: string, direction: SortDirection) =>
+        onSortTable(column as SortColumn, direction)
```

Comment:
Trying to think of a way to do this without having to assert here. IndexesTable doesn't know about the specific column types and it is just data at that point.